language/bytecode_verifier/src/check_duplication.rs

```diff
@@ -181,12 +181,6 @@ impl<'a> DuplicationChecker<'a> {
                 idx,
                 StatusCode::INCONSISTENT_FIELDS,
             ));
-        } else if start_field_index != self.module.field_defs().len() {
```

Comment:
can we add this back?
I understand that it will be checked twice in some scenario and I also understand we may not want to check this anymore but it feels it may be good to leave everything as it was.
Thoughts?

Reply:
I'd be prone to have it removed. Here's my reasoning:
1) All tests pass with it removed => this means don't have any input/test that touches this.
2) To add tests for this, we can just move the unused checker to the linter crate and add the tests there. It's very little work to define the public interfaces.
3) Better to remove it for consistency. Why only unused fields are checked, but not entries in the other pools?

Reply:
I figured we are not generating any module with that problem and it's fine to go in like this too.
To me that check is more about invalid entries rather than unused ones.
But I am fine taking that discussion in a broader context